Library/Homebrew/os/mac/mach.rb

```diff
@@ -51,8 +51,10 @@ def mach_data
     end
   end
 
-  def dynamically_linked_libraries
-    macho.linked_dylibs
+  def dynamically_linked_libraries(except: :none)
+    lcs = macho.dylib_load_commands.reject { |lc| lc.type == except }
+
+    lcs.map(&:name).map(&:to_s)
```

Comment:
`lcs.map(&:name).map(&:to_s)` should be `lcs.map(&:name).map(&:to_s).uniq` otherwise universal libraries show up twice. This then triggers a failure to bottle universal binaries & libraries, like my wine PR (https://github.com/Homebrew/homebrew-core/pull/10532 ; test results: https://bot.brew.sh/job/Homebrew%20Core/17442/version=sierra/testReport/junit/brew-test-bot/sierra/bottle_wine/)

Before the commit, this code relied on `macho.linked_dylibs`, which does apply `uniq`:

```
    def linked_dylibs
      # Individual architectures in a fat binary can link to different subsets
      # of libraries, but at this point we want to have the full picture, i.e.
      # the union of all libraries used by all architectures.
      machos.map(&:linked_dylibs).flatten.uniq
    end
```